### ISSUE TITLE
Any known mirrors for this?

### ISSUE THREAD

**Issue by dominykas** [dodo/node-unicodetable#16]
www.unicode.org seems to be down at the moment - are there alternate ways or any known mirrors for UnicodeData.txt?

**Comment by plato-cambrian** [comment]
+1 why not just commit this file in the package? it seems very fragile that any project that depends on `unicode` will fail to install if unicode.org is down

**Comment by dodo** [comment]
I don't like the idea of putting dependencies into source controls.
Also there is still the possibility of installing `UnicodeData.txt` on [your system](https://github.com/dodo/node-unicodetable/blob/0.6.0/install.js#L15-16) by installing `unicode-data` via your system package installer.

Please let me know if this working out for you and/or your distribution is missing.

**Comment by ltanme** [comment]
try to read file /usr/share/unicode/UnicodeData.txt …
/usr/share/unicode/UnicodeData.txt not found.
try to read file /usr/share/unicode-data/UnicodeData.txt …
/usr/share/unicode-data/UnicodeData.txt not found.
try to read file UnicodeData.txt …
UnicodeData.txt not found.
try to download …
GET unicode.org:80/Public/UNIDATA/UnicodeData.txt
request timed out.

mac pro os x node install.js not work

**Comment by sjanecki** [comment]
same thing again
could you do something with this?

```
curl -vvv -s -i 'http://unicode.org:80/Public/UNIDATA/UnicodeData.txt'
* About to connect() to unicode.org port 80 (#0)
*   Trying 216.97.88.9... Connection timed out
* couldn't connect to host
* Closing connection #0
```

**Comment by jakobloekke** [comment]
I found this: http://opensource.apple.com/source/Heimdal/Heimdal-247.7/lib/wind/UnicodeData.txt?txt

Can I somehow use that instead?

**Comment by tahiryasin** [comment]
Run `sudo apt-get install unicode-data` and everything will be good.

**Comment by srl295** [comment]
Unicode throttles excess Downloads.

**Comment by vladinator1000** [comment]
Got a problem with the same error on windows, can't install anything in node:
`error C:\Projects\Web\yupty-api\node_modules\unicode: Command failed.
Exit code: 1
Command: C:\WINDOWS\system32\cmd.exe
Arguments: /d /s /c node install.js
Directory: C:\Projects\Web\yupty-api\node_modules\unicode
Output:
try to read file /usr/share/unicode/UnicodeData.txt …
/usr/share/unicode/UnicodeData.txt not found.
try to read file /usr/share/unicode-data/UnicodeData.txt …
/usr/share/unicode-data/UnicodeData.txt not found.
try to read file UnicodeData.txt …
UnicodeData.txt not found.
try to download …
GET unicode.org:80/Public/UNIDATA/UnicodeData.txt
Error while downloading UnicodeData.txt: Error: connect ETIMEDOUT 216.97.88.9:80
Please download file manually, put it next to the install.js file and run `node install.js` again.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.`

**Comment by srl295** [comment]
OK… experimental, can you try https://sf.net/p/unicode-org/files  ? Specifically https://sourceforge.net/projects/unicode-org/files/UNIDATA/UnicodeData.txt 

- this is experimental 🚧 (meaning: could break at any time) so… don't depend on it for now. Let me know if it seems to work.

**Comment by tdanecker** [comment]
fixed in 9.0.0 (https://github.com/eversport/node-unicodetable/tree/9.0.0): the UnicodeData.txt isn't required any more as the category files are included directly.

**Comment by srl295** [comment]
@tdanecker I'm still new to looking at this module. I guess it's OK to look at `/usr/share/unicode*` but I'd wonder how old that file is. on debian sid it's at 9.0 (current) but it might be older elsewhere.

**Comment by tdanecker** [comment]
@srl295 it's only used to generate the category/*.js files, which is now done before the package is published. I didn't really clean up the code in the install.js file yet, but it isn't executed at install time anymore, so only those who publish the package have to check, that the UnicodeData.txt is the current one. I intend to match the package version with it, so when using the version 9.0.0 of the unicode npm package, you should always get the 9.0 unicode data ;)

**Comment by sdesalas** [comment]
Guys, there is a workaround for this that is platform agnostic (works on windows too!), its targeted at situations where other projects depend on this package in CI.

The [`install.js`](https://github.com/dodo/node-unicodetable/blob/master/install.js) process looks for an environment variable called `NODE_UNICODETABLE_UNICODEDATA_TXT` that contains the local path for the `UnicodeData.txt` file before going to `unicode.org` to look for it, so if you set this it will prevent the remote connection errors from happening.

For example:
 
1. Copy `UnicodeData.txt` into a `UnicodeData` folder at the root of your project.
2. Add following to your CI before `npm install`.

```
# Windows
> set NODE_UNICODETABLE_UNICODEDATA_TXT=%CD%\UnicodeData\UnicodeData.txt
# Unix / Linux / OSX
$ export NODE_UNICODETABLE_UNICODEDATA_TXT=$PWD/UnicodeData/UnicodeData.txt
```

Hope this helps!

**Comment by kriscarle** [comment]
Ha, 10 days later and my IP is still blocked by unicode.org. That seems a little excessive, not exactly "throttling"?  Has anyone reached out to them to let them know this is being fixed? Maybe someone can donate some CDN hosting for them to redirect this traffic to until everyone has upgraded?

**Comment by srl295** [comment]
@kriscarle write me (my email is on my profile)  … and did you try the experimental mirror mentioned in https://github.com/dodo/node-unicodetable/issues/16#issuecomment-291533974 ?

**Comment by dominykas** [comment]
stalebot-ing myself.

**Comment by srl295** [comment]
@dominykas ok…  any comments on the experimental mirrors mentioned?

**Comment by dominykas** [comment]
Sorry, this is no longer relevant for me, I opened this 3 years ago and this repo hasn't had updates since, so if anyone still depends on this, I'd say the correct approach is to fork and figure it out.
